api/node/server/server.go

```diff
@@ -0,0 +1,198 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/oapi-codegen/runtime/strictmiddleware/nethttp"
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/activation"
+	"github.com/spacemeshos/go-spacemesh/api/node/models"
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/p2p/pubsub"
+)
+
+//go:generate mockgen -typed -package=server -destination=mocks.go -source=server.go
+
+type poetDB interface {
+	ValidateAndStore(ctx context.Context, proofMessage *types.PoetProofMessage) error
+}
+
+type Server struct {
+	atxService activation.AtxService
+	publisher  pubsub.Publisher
+	poetDB     poetDB
+	logger     *zap.Logger
+}
+
+var _ StrictServerInterface = (*Server)(nil)
+
+func NewServer(
+	atxService activation.AtxService,
+	publisher pubsub.Publisher,
+	poetDB poetDB,
+	logger *zap.Logger,
+) *Server {
+	return &Server{
+		atxService: atxService,
+		publisher:  publisher,
+		poetDB:     poetDB,
+		logger:     logger,
+	}
+}
+
+func (s *Server) IntoHandler(mux *http.ServeMux) http.Handler {
+	loggingMid := func(f nethttp.StrictHTTPHandlerFunc, operationID string) nethttp.StrictHTTPHandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, req any) (any, error) {
+			uuid := uuid.New()
+			s.logger.Debug(
+				"request",
+				zap.Stringer("request_id", uuid),
+				zap.String("operation", operationID),
+				zap.Any("request", req),
+			)
+			response, err := f(ctx, w, r, req)
+			s.logger.Debug(
+				"response",
+				zap.Stringer("request_id", uuid),
+				zap.String("operation", operationID),
+				zap.Any("response", response),
```

Comment:
so `response` doesn't always actually serialize into anything, because it is an interface. i think this field can be removed.

Reply:
Yes, but `zap.Any()` uses reflection to log the exported fields.

Reply:
from the logs:
```
node-service-1        | 2024-10-31T14:42:36.796Z	DEBUG	node.atx-service	response	{"request_id": "1846e926-482f-463c-80f6-46ccbe1d9eea", "operation": "GetActivationLastAtxNodeId", "response": {}}
```
Since `response` which is returned from the handler is an `interface{}`, the part of `zap` using reflection only makes sense when the returned type actually has exported fields if I understand correctly. I would find it more meaningful to find out the http status code here, rather than an inlining of the actual data type (which might not make much sense anyway)